src/translator/terminology.cpp

```diff
@@ -0,0 +1,50 @@
+#include "terminology.h"
+
+#include <sstream>
+
+namespace marian {
+namespace bergamot {
+
+namespace {
+
+struct Hit {
+  size_t pos;
+  TerminologyMap::const_iterator entry;
+};
+
+}  // namespace
+
+std::string ReplaceTerminology(std::string const &str, TerminologyMap const &terminology) {
```

Comment:
Once `TerminologyMap` becomes very large, this method is going to be stupidly slow. At that point we should change it into a [state machine type of search thingy](https://en.wikipedia.org/wiki/Commentz-Walter_algorithm).

Reply:
If terminology map is very large, i feel something wrong is happening.
State machine would always be better but also what is the time this would take.